doc/rbd/rbd-openstack.rst

```diff
@@ -128,9 +128,9 @@ Setup Ceph Client Authentication
 If you have `cephx authentication`_ enabled, create a new user for Nova/Cinder
 and Glance. Execute the following::
 
-    ceph auth get-or-create client.glance mon 'profile rbd' osd 'profile rbd pool=images'
-    ceph auth get-or-create client.cinder mon 'profile rbd' osd 'profile rbd pool=volumes, profile rbd pool=vms, profile rbd-read-only pool=images'
-    ceph auth get-or-create client.cinder-backup mon 'profile rbd' osd 'profile rbd pool=backups'
+    ceph auth get-or-create client.glance mon 'profile rbd' osd 'profile rbd pool=images' mgr 'profile rbd with pool=images'
+    ceph auth get-or-create client.cinder mon 'profile rbd' osd 'profile rbd pool=volumes, profile rbd pool=vms, profile rbd-read-only pool=images' mgr 'profile rbd with pool=volumes, profile rbd with pool=vms'
+    ceph auth get-or-create client.cinder-backup mon 'profile rbd' osd 'profile rbd pool=backups' mgr 'profile rbd with pool=backups'
```

Comment:
I understand the reasons why it is so, but I suppose it will be a source of confusion for users that when specifying a pool in a profile for `osd` they should not use `with` keyword but for mgr they should. I can imagine users erroneously adding `with` to osd or not adding for mgr, and blaming developers for adding yet another inconsistency.

May be add a special handling for `pool` and `namespace` keys in mgr profile, so users could specify it without `with` keyword? 

For module I think `with` is ok, it is only profile that concerns me a little.

Reply:
The good news is that "ceph auth" will fail if they erroneously add or skip the `with` clause. Since the MGR constraints are free-form and not syntax literals, I think we would really want to keep the `with` to provide an anchor to the syntax. Perhaps instead we can add an optional `with` clause to the OSD 'profile XYZ' cap?

Reply:
... actually, been playing around w/ it and it doesn't seem like the `with` is really necessary to ensure the grammar can correctly parse the subsequent clauses. I was worried about someone just running `profile rbd network=xyx" and having it parse as a key/value, but I think the best solution would be to actually validate the keywords and drop the cap if they aren't valid.

Reply:
Actually I meant to have  only two special cases for pool and namespace and allow to specify other arbitrary keys only with `with ...`. But I like your current version too (probably more).

Reply:
Ah -- yeah, I didn't want to lock in 'pool' or 'namespace' as globally valid profile keys since it might not apply to other things (e.g. CephFS might lock down to volume / sub-volume)